watertap/unit_models/nanofiltration_DSPMDE_0D.py

```diff
@@ -0,0 +1,1394 @@
+                            - 1.56034 * b.lambda_comp[t, j]
+                            + 0.528155 * b.lambda_comp[t, j] ** 2
+                            + 1.91521 * b.lambda_comp[t, j] ** 3
+                            - 2.81903 * b.lambda_comp[t, j] ** 4
+                            + 0.270788 * b.lambda_comp[t, j] ** 5
+                            + 1.10115 * b.lambda_comp[t, j] ** 6
+                            - 0.435933 * b.lambda_comp[t, j] ** 7) /
+                           (1 - b.lambda_comp[t, j] + eps) ** 2,
+                           )
+                # Relationship used by Geraldes & Alves
+                # (1
+                #     - 2.3 * b.lambda_comp[t, j]
+                #     + 1.154 * b.lambda_comp[t, j] ** 2
+                #     + 0.224 * b.lambda_comp[t, j] ** 3
+                #     )
+
+
+        @self.Expression(self.flowsheet().config.time,
+                         solute_set,
+                         doc="Pore diffusion coefficient")
+        def diffus_pore_comp(b, t, j):
+            return b.hindrance_factor_diffusive_comp[t, j] * b.feed_side.properties_in[t].diffus_phase_comp['Liq', j]
+
+        @self.Expression(self.flowsheet().config.time,
+                         solute_set,
+                         doc="Convective hindered transport coefficient")
+        def hindrance_factor_convective_comp(b, t, j):
+            return ((1 + 3.867 * b.lambda_comp[t, j]
+                     - 1.907 * b.lambda_comp[t, j] ** 2
+                     - 0.834 * b.lambda_comp[t, j] ** 3)
+                    / (1 + 1.867 * b.lambda_comp[t, j] - 0.741 * b.lambda_comp[t, j]**2))
+
+        @self.Expression(self.flowsheet().config.time,
+                         solute_set,
+                         doc="Steric partitioning factor")
+        def partition_factor_steric_comp(b, t, j):
+            return (1 - b.lambda_comp[t, j]) ** 2
+
+        @self.Expression(self.flowsheet().config.time,
+                         solute_set,
+                         doc="Gibbs free energy of solvation for each ion")
+        def gibbs_solvation_comp(b, t, j):
+            return (b.feed_side.properties_in[t].charge_comp[j] ** 2
+                    * Constants.elemental_charge ** 2
+                    / (8 * Constants.pi
+                       * Constants.vacuum_electric_permittivity
+                       * b.feed_side.properties_in[t].radius_stokes_comp[j])
+                    * (1 / b.feed_side.properties_in[t].dielectric_constant - 1 / b.dielectric_constant_pore[t]))
+
+        @self.Expression(self.flowsheet().config.time,
+                         solute_set,
+                         doc="Born solvation contribution to partitioning")
+        def partition_factor_born_solvation_comp(b, t, j):
+            return (-b.gibbs_solvation_comp[t, j]
+                    / (Constants.boltzmann_constant * b.feed_side.properties_in[t].temperature))
+
+    def initialize(
```

Comment:
In order to facilitate initialization of costing methods, IDAES has changed how initialization is called. There is now a common, centralized `initialize` method in the base classes which calls to a model specific `initialize_build` method (i.e. this method). This way, the central method can handle determining what to do with "add-ons" like costing, and also handles the final solve of the full model (including raising an `InitializationError` if the final step fails).

Rather than overload `initialize`, you should instead just rename this to `initialize_build` and everything should still work.